src/Agent.js

```diff
@@ -58,11 +58,12 @@ export default class Agent {
 	 * @param {String} query         Query parameters
 	 * @param {Object} form          Form fields
 	 * @param {Object} files         array of file names and file content
+	 * @parma {Object} context       the invocation context, describing the tool and project.
 	 * @return {Promise} A promise. fulfilled with {body, statusCode}, rejected with { statusCode, errorDescription, error, body }
 	 */
-	request({ uri, method, data = undefined, auth, query = undefined, form = undefined, files = undefined }) {
+	request({ uri, method, data = undefined, auth, query = undefined, form = undefined, files = undefined, context = undefined }) {
```

Comment:
Since you're changing this method I think now would be a good time to remove the `= undefined` for each parameter. That doesn't add anything to the implementation since parameters that are omitted are `undefined` by default.